component/main.jsonnet

```diff
@@ -1,10 +1,207 @@
 // main template for appuio-cloud-reporting
+local alerts = import 'alerts.libsonnet';
+local common = import 'common.libsonnet';
 local kap = import 'lib/kapitan.libjsonnet';
 local kube = import 'lib/kube.libjsonnet';
+local netPol = import 'networkpolicies.libsonnet';
 local inv = kap.inventory();
 // The hiera parameters for the component
 local params = inv.parameters.appuio_cloud_reporting;
 
+local monitoringLabel =
+  if std.startsWith(inv.parameters.facts.distribution, 'openshift') then
+    {
+      'openshift.io/cluster-monitoring': 'true',
+    }
+  else
+    {
+      SYNMonitoring: 'main',
+    };
+
+local formatImage = function(ref) '%(registry)s/%(repository)s:%(tag)s' % ref;
+
+local escape = function(str)
+  std.join('',
+           std.map(
+             function(c)
+               if ((c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z') || (c >= '0' && c <= '9')) then c else '-'
+             , str
+           ));
+
+local dbSecret = kube.Secret('reporting-db') {
+  metadata+: {
+    namespace: params.namespace,
+    labels+: common.Labels,
+  },
+  stringData: {
+    username: params.database.username,
+    password: params.database.password,
+  },
+};
+
+local promURLSecret = kube.Secret('prom-url') {
+  metadata+: {
+    namespace: params.namespace,
+    labels+: common.Labels,
+  },
+  stringData: {
+    url: params.prometheus.url,
+  },
+};
+
+local erpURLSecret = kube.Secret('odoo-url') {
+  metadata+: {
+    namespace: params.namespace,
+    labels+: common.Labels,
+  },
+  stringData: {
+    url: params.erp_adapter.url,
+  },
+};
```

Comment:
These could be configmaps, unless we optionally support credentials in the URLs.

Reply:
The Odoo url does contain secrets. The prometheus one not (yet)?